mitodl/mit-open
# Sloan Executive Education blog ETL

### What are the relevant tickets?
Closes #611 

### Description (What does it do?)
Adds an ETL pipeline for news from the Sloan Executive Education blog


### How can this be tested?
- Run `./manage.py backpopulate_news_events --pipeline sloan_exec_news_etl`.
When it is done, there should be ~20 news items ingested from that source (you can see them at http://localhost:8063/admin/news_events/feeditem/, filter by source)
- Make sure the data looks reasonable and the links work.


### Additional Context
The blog html is generated entirely via javascript.  To get the item data, the code has to use `requests.Session` to first get the index page along with cookies, then search for certain variable values like `fwuid` in the response and substitute those into a POST request to grab the actual news articles.
